conversation/backend/src/main/java/org/entcore/conversation/util/FolderUtil.java

```diff
@@ -50,72 +39,63 @@ public class FolderUtil {
      * @return a JsonArray representing the root folders of the hierarchical tree structure.
      */
     static public JsonArray listToTree(final JsonArray list, final int depth) {
-        // Sort root folders (with depth=1) by ID
-        SortedSet<JsonObject> rootFolders =  new TreeSet<>(compareOnIdAsc);
-        // Sort children folders (with depth=2+) by ID.
-        SortedSet<JsonObject>[] folderLevelById = new SortedSet[depth];
-        // Also sort children folders (with depth=2+) by parent folder ID
-        SortedSet<JsonObject>[] folderLevelByParentId = new SortedSet[depth];
+        // Sort the list, case-insensitive, but capitals are prioritized
+        List<JsonObject> sortedList = list.stream()
+            .map(obj -> (JsonObject) obj)
+            .sorted(Comparator.comparing(
+                o -> o.getString(NAME),
+                (s1, s2) -> {
+                    int compare = s1.compareToIgnoreCase(s2);
+                    return compare != 0 ? compare : s1.compareTo(s2);
+                }
+            ))
+            .collect(Collectors.toList());
 
-        // Sets of level 0 (for depth=1) are not initialized, because only `rootFolders` will be used for level 0.        
-        for( int level=1; level<depth; level++ ) {
-            folderLevelById[level] = new TreeSet<>(compareOnIdAsc);
-            folderLevelByParentId[level] = new TreeSet<>(compareOnParentThenIdAsc);
+        // Indexing by id
+        Map<String, JsonObject> itemsById = new HashMap<>();
+        for (JsonObject item: sortedList) {
+            itemsById.put(item.getString(ID), item.copy());
         }
-        // Put every folder in the SortedSet dedicated to its depth.
-        list.stream().forEach((item) -> {
-            if( ! (item instanceof JsonObject) ) return;
-            final JsonObject folder = (JsonObject) item;
-            final int folderLevel = (int) (getDepth(folder) - 1);
-            if(folderLevel==0) {
-                rootFolders.add(folder);
-            } else if(0<folderLevel && folderLevel<depth) {
-                folderLevelById[folderLevel].add(folder);
-                folderLevelByParentId[folderLevel].add(folder);
+
+        // Indexing children
+        Map<String, List<JsonObject>> childrenMap = new HashMap<>();
+        for (JsonObject item: sortedList) {
```

Comment:
à rapatrier dans la boucle `for (JsonObject item: sortedList)` de la ligne 56.
=> on essaie d'éviter d'itérer plusieurs fois sur la même collection, lorsque c'est possible.